source/postprocess/visualization/theta_tauisa.cc

```diff
@@ -0,0 +1,215 @@
+
+        // Loop over cells and calculate theta in each one
+        // Note that we start after timestep 0 because we need the strain rate,
+        // which doesn't exist during the initial step
+        unsigned int cell_index = 0;
+        for (; cell != endc; ++cell, ++cell_index)
+          {
+            if (cell->is_locally_owned() && this->get_timestep_number() > 0)
+              {
+
+                // Fill the material model objects for the cell (for strain rate)
+                fe_values.reinit(cell);
+                in.reinit(fe_values, cell, this->introspection(),
+                          this->get_solution(), true);
+                // Also get velocities and velocity gradients
+                std::vector<Tensor<1, dim> > velocities(n_q_points,
+                                                        Tensor<1, dim>());
+                std::vector<Tensor<2, dim> > velocity_gradient(n_q_points,
+                                                               Tensor<2, dim>());
+                fe_values[this->introspection().extractors.velocities].get_function_values(
+                  this->get_solution(), velocities);
+                fe_values[this->introspection().extractors.velocities].get_function_gradients(
+                  this->get_solution(), velocity_gradient);
+
+                // Calculate eigenvalues of strain rate and take maximum (absolute value)
+                // to get tauISA, the timescale for grain rotation toward the infinite strain axis
+                const SymmetricTensor<2, dim> strain_rate = in.strain_rate[0];
+                std::array<double, dim> strain_rate_eigenvalues = eigenvalues(
+                                                                    strain_rate);
+                double lamda1 = std::max(std::abs(strain_rate_eigenvalues[0]),
+                                         std::abs(strain_rate_eigenvalues[dim]));
+                double tauISA = 1.0 / lamda1;
+
+                // Get the deformation gradient tensor in the limit of "infinite" time
+                // from the velocity gradient tensor. As per Kaminski and Ribe (2002),
+                // tmax = 75*tauISA is infinite enough for most applications.
+                unsigned int maxorder = 4;
+                Tensor<2, dim> F = Tensor<2, dim>(
+                                     unit_symmetric_tensor<dim, double>());
+                double tmax = 75 * tauISA;
+
+                for (unsigned int i = 0; i < maxorder; ++i)
+                  {
+                    Tensor<2, dim> multiplier = Tensor<2, dim>(
+                                                  unit_symmetric_tensor<dim, double>());
+                    double factor = 1.0;
+                    for (unsigned int j = 0; j < i; ++j)
+                      {
+                        multiplier = multiplier
+                                     * Tensor<2, dim>(velocity_gradient[0]);
+                        factor *= tmax / (j + 1);
```

Comment:
I'm not sure I understand this algorithm. Can you add a comment that explain what this does?